pennylane/_qubit_device.py

```diff
@@ -24,6 +24,7 @@
 import warnings
 
 import numpy as np
+from jax.errors import TracerArrayConversionError
```

Comment:
@antalszava, 
This line seems to be causing the tests to fail as they don't install jax into the environment. Any ideas on how we should proceed? I need this error type so that I can catch it and act accordingly in the logic below.

Reply:
@Jaybsoni could the entailed logic be added to `qml.math.squeeze`? We wouldn't like to add `jax` based imports into this file because this file shouldn't require users to have JAX installed. One last resort would be to try importing JAX, but let's try to circumvent that first and delegate any logic as such to `qml.math`.

Reply:
Good idea! 
I will give that a shot !

Reply:
@antalszava, I shifted the logic into `qml.math.squeeze`. The tests will still fail however because jax isn't installed in the virtual environment where we are running these tests. Can I go ahead and change that? Or do you have some other suggestion?

Reply:
Ahh it might be because jax isn't a PL dependancy so it doesn't get installed in the test environment ?

Reply:
Yes. Jax is only imported by one of the CIs and should be conditionally imported within the code too, e.g.,
```python
    interface = _multi_dispatch(values)
    if interface == "jax":
        import jax
        # do something here
```
Other occurrences of importing jax within `qml.math` can help here.